subsys/nrf_security/src/drivers/Kconfig

```diff
@@ -4,56 +4,217 @@
+	depends on PSA_WANT_ALG_SHA_1 || \
+			   PSA_WANT_ALG_SHA_224 || \
+			   PSA_WANT_ALG_SHA_256
+	help
+	  Enable CryptoCell for the PSA hash APIs. Enabling this
+	  will provide support for SHA1, SHA224 and SHA256.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_KEY_PAIR
+config PSA_USE_CC3XX_MAC_DRIVER
+	prompt "Enable CryptoCell for the PSA mac APIs"
 	bool
 	default y
-	depends on PSA_WANT_KEY_TYPE_RSA_KEY_PAIR && \
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_CMAC || \
+			   PSA_WANT_ALG_HMAC
+	help
+	  Enable CryptoCell for the PSA mac APIs. Enabling this
+	  will provide support for HMAC and CMAC.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_PUBLIC_KEY
+config PSA_USE_CC3XX_ECDH_DRIVER
+	prompt "Enable CryptoCell for the PSA key agreement APIs"
 	bool
 	default y
-	depends on PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY && \
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_ECDH
+	help
+	  Enable CryptoCell for the PSA key agreement APIs. Enabling this
+	  will provide support for key agreement with ECDH.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_AES
+config PSA_USE_CC3XX_SIGNATURE_DRIVER
+	prompt "Enable CryptoCell for the PSA asymmetric signature APIs"
 	bool
 	default y
-	depends on PSA_WANT_KEY_TYPE_AES && \
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_ECDSA || \
+			   PSA_WANT_ALG_RSA_PKCS1V15_SIGN || \
+			   PSA_WANT_ALG_RSA_PSS
+	help
+	  Enable CryptoCell for the PSA asymmetric signature APIs. Enabling this
+	  will provide support signing/verification with ECDSA in deterministic
+	  and randomized modes and RSA in PKCS1V15 and PSS modes.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_CHACHA20
+config PSA_USE_CC3XX_ASYMMETRIC_DRIVER
+	prompt "Enable CryptoCell for the PSA asymmetric encryption APIs"
 	bool
 	default y
-	depends on PSA_WANT_KEY_TYPE_CHACHA20 && \
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_RSA_PKCS1V15_CRYPT || \
+			   PSA_WANT_ALG_RSA_OAEP
+	help
+	  Enable CryptoCell for the PSA asymmetric encryption APIs. Enabling this
+	  will provide support for assymetric encryption with RSA
+	  in PKCS1V15 and OAEP modes.
 
+config PSA_USE_CC3XX_KEY_PAIR_DRIVER
+	prompt "Enable CryptoCell for the PSA key management APIs"
+	bool
+	default y
+	depends on PSA_USE_CC3XX_ASYMMETRIC_DRIVER || \
+			   PSA_USE_CC3XX_SIGNATURE_DRIVER  || \
+			   PSA_USE_CC3XX_ECDH_DRIVER
+	help
+	  Enabling this will provide support for key generation, key importing and
+	  public exporting for asymmetric keys. ECC and RSA keys are supported.
 
-rsource "nrf_cc3xx/Kconfig"
+endif
+
+config PSA_USE_CC3XX_CTR_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_USE_CTR_DRBG_DRIVER
+	depends on NRF_CC3XX_PLATFORM || BUILD_WITH_TFM
 
-rsource "nrf_cc3xx_platform/Kconfig"
+config PSA_USE_CC3XX_HMAC_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_USE_HMAC_DRBG_DRIVER
+	depends on NRF_CC3XX_PLATFORM || BUILD_WITH_TFM
```

Comment:
Ah, is NRF_CC3XX_PLATFORM false for non-secure?

Reply:
Exactly, that's the rationale.

Reply:
Is there something we can && this with to hide it as an optoin on non-CC3XX platforms?

e.g.

NRF_CC3XX_PLATFORM || (BUILD_WITH_TFM && HAS_CRYPTOCELL)

?

Reply:
It is noisy for users to see options that are not applicable to their platform.

Reply:
CRYPTOCELL_USABLE is in the process being removed. This was one of the types of aggregated configurations that could simplify depends on in cases like this. Additionally we are working on DTS changes for CryptoCell to make CryptoCell be "available" also in NS scope, and I'm sure we can use something like that to simplify dependency tracking.  @joerchan can chime in here.

Reply:
I can update it to CRYPTOCELL_USABLE since it is equivalent indeed.   I agree with you Joakim that's it's better to not include in this PR linking improvements for the platform library.